task_02/novikov_dmitry/task_02.cpp

```diff
@@ -0,0 +1,242 @@
+#include <iostream>
+#include <vector>
+#include <string>
+#include <map>
+#include <memory>
+#include <string>
+#include <fstream>
+#include <sstream>
+#include <set>
+
+class Vertex {
+private:
+    int vertex_id;
+public:
+    //other information
+    Vertex() : vertex_id(-1) {}
+    Vertex(int id) : vertex_id(id) {}
+    Vertex(const Vertex &item) : vertex_id(item.vertex_id) {}
+    //Vertex(int id, ...) {}
+    bool operator >(const Vertex &v) const {
+        return this->vertex_id > v.vertex_id;
+    }
+    bool operator <(const Vertex &v) const {
+        return this->vertex_id < v.vertex_id;
+    }
+    bool operator ==(const Vertex &v) const {
+        return this->vertex_id == v.vertex_id;
+    }
+    bool operator !=(const Vertex &v) const {
+        return this->vertex_id != v.vertex_id;
+    }
+
+    std::string vertex_to_str() const{

```

Comment:
Опять же, мы итак знаем что это `vertex`, ненужное уточнение. Представь, как это будет выглядеть в коде:
```cpp
auto vertex = Vertex();
vertex.vertex_to_str();
```
```suggestion
    std::string to_str() const {
```